src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentGenericTypePass.cs

```diff
@@ -410,7 +409,7 @@ private void RewriteTypeNames(TypeNameRewriter rewriter, ComponentIntermediateNo
 
             foreach (var childContent in node.ChildContents)
             {
-                if (childContent.BoundAttribute?.IsGenericTypedProperty() ?? false && childContent.TypeName != null)
+                if ((childContent.BoundAttribute?.IsGenericTypedProperty() ?? false) && childContent.TypeName != null)
```

Comment:
We can see that on line 416 we use the `TypeName` following that through, it would eventually throw a NullRef exception, so we can be sure that actually performing this check isn't changing behavior.

```suggestion
                if (childContent.BoundAttribute?.IsGenericTypedProperty() == && childContent.TypeName != null)
```
As above